src/MSBuildLocator/DotNetSdkLocationHelper.cs

```diff
@@ -130,30 +131,65 @@ private static void ModifyUnmanagedDllResolver(Action<AssemblyLoadContext> resol
 
         private static IntPtr HostFxrResolver(Assembly assembly, string libraryName)
         {
-            var hostFxrLibName = "libhostfxr";
-            var libExtension = RuntimeInformation.IsOSPlatform(OSPlatform.OSX) ? "dylib" : "so";
+            string hostFxrLibName = "libhostfxr";
+            string libExtension = RuntimeInformation.IsOSPlatform(OSPlatform.OSX) ? "dylib" : "so";
 
-            if (!hostFxrLibName.Equals(libraryName))
+            // the DllImport hardcoded the name as hostfxr.
+            if (!hostFxrLibName.Equals(libraryName, StringComparison.OrdinalIgnoreCase) && !libraryName.Equals("hostfxr", StringComparison.OrdinalIgnoreCase))
```

Comment:
nit: If `libraryName` we get here comes from the `DllImport` attribute verbatim, why not check for it with a straight equals? I.e. no need for case insensitivity or detecting the other name.

```suggestion
            // the DllImport hardcoded the name as hostfxr.
            if (!libraryName.Equals("hostfxr"))
```

Reply:
fair point. I just don't want to change the original behavior, but since it is a recent change, I guess no other logic would depend on the original path.  I will do that, if it works.

Reply:
yeah, Christian verified it, so I deleted that piece.